src/main/java/com/iexec/worker/executor/TaskNotificationService.java

```diff
@@ -204,11 +203,17 @@ protected void onTaskNotification(TaskNotification notification) {
 
     }
 
-    private boolean storeWorkerpoolAuthorizationFromExtraIfPresent(TaskNotificationExtra extra) {
-        if (extra != null && extra.getWorkerpoolAuthorization() != null){
-            return contributionService.putWorkerpoolAuthorization(extra.getWorkerpoolAuthorization());
+    private boolean storeWorkerpoolAuthAndSmsFromExtraIfPresent(TaskNotificationExtra extra) {
+        boolean success = true;
+        if(extra != null && extra.getWorkerpoolAuthorization() != null){
+            success = workerpoolAuthorizationService
```

Comment:
What happens if `workerpoolAuthorizationService.putWorkerpoolAuthorization(...)` fails? Are we sure we need/want to attach the SMS URL to the task?